### PR TITLE
Use Identity package reference

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
Event Hubs bits look good to me.

**Review by JoshLove-msft — COMMENTED** [review]
Can you rebase now that the Files.Shares rename is merged?

File: sdk/storage/Azure.Storage.Files.Shares/samples/Azure.Storage.Files.Shares.Samples.Tests.csproj [hunk 1]
```diff
@@ -5,9 +5,10 @@
     <IsPackable>false</IsPackable>
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Azure.Identity" />
+    <ProjectReference Include="$(MSBuildThisFileDirectory)..\src\Azure.Storage.Files.csproj" />
```
Comment (JoshLove-msft) [inline sdk/storage/Azure.Storage.Files.Shares/samples/Azure.Storage.Files.Shares.Samples.Tests.csproj]: This should be deleted.

**Review by tg-msft — APPROVED** [review]

File: sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj [hunk 2]
```diff
@@ -6,6 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Azure.Identity" />
```
Comment (heaths) [inline sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj]: Shouldn't Azure.Core.Tests still use a ProjectReference?

**Review by heaths — APPROVED** [review]
Besides the other comment I added, wouldn't it make sense for tests and samples - or maybe just tests - to use the same package-or-project reference as the assembly under test? That way, should we ever need to coordinate dependent changes, we can - at least in the inner dev loop - use a single build variable to switch between the two.

File: sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj (hunk 2 above)
Comment (pakrym) [reply sdk/core/Azure.Core/tests/Azure.Core.Tests.csproj]: Not to Azure.Identity

**Review by AlexGhiondea — APPROVED** [review]

**Review by schaabs — APPROVED** [review]

**Comment by danieljurek** [comment]
The fact that `net - client - ci` is showing yellow here even though it's already passed is **not blocking**. I'm OK with us merging this.

**Pull request merged**